src/components/application_manager/test/application_manager_impl_test.cc

```diff
@@ -2018,6 +2018,214 @@ TEST_F(
   EXPECT_EQ(1u, app_list.size());
 }
 
+TEST_F(
+    ApplicationManagerImplTest,
+    RegisterApplication_WEPNonMediaAppRegisterActivateDeactivateExit_EXPECT_CORRECT_HMI_STATES) {
+  EXPECT_CALL(
+      mock_session_observer_,
+      GetDataOnSessionKey(kConnectionKey,
+                          _,
+                          _,
+                          testing::An<connection_handler::DeviceHandle*>()))
+      .WillOnce(DoAll(SetArgPointee<3u>(kDeviceId), Return(0)));
+
+  smart_objects::SmartObject rai(smart_objects::SmartType_Map);
+  smart_objects::SmartObject& params = rai[strings::msg_params];
+  params[strings::app_id] = kAppId;
+  params[strings::language_desired] = mobile_api::Language::EN_US;
+  params[strings::hmi_display_language_desired] = mobile_api::Language::EN_US;
+
+  rai[strings::params][strings::connection_key] = kConnectionKey;
+  rai[strings::msg_params][strings::app_name] = kAppName;
+  rai[strings::msg_params][strings::app_hmi_type] = "WEB_VIEW";
+  rai[strings::msg_params][strings::is_media_application] = false;
+  rai[strings::msg_params][strings::sync_msg_version][strings::minor_version] =
+      APIVersion::kAPIV2;
+  rai[strings::msg_params][strings::sync_msg_version][strings::major_version] =
+      APIVersion::kAPIV6;
+
+  rai[strings::params][strings::protocol_version] =
+      protocol_handler::MajorProtocolVersion::PROTOCOL_VERSION_2;
+
+  const auto rai_ptr = std::make_shared<smart_objects::SmartObject>(rai);
+
+  std::unique_ptr<plugin_manager::RPCPluginManager> rpc_plugin_manager(
+      new MockRPCPluginManager());
+  app_manager_impl_->SetPluginManager(rpc_plugin_manager);
+  auto wep_app = app_manager_impl_->RegisterApplication(rai_ptr);
+  wep_app->set_is_media_application(false);
+
+  EXPECT_EQ(protocol_handler::MajorProtocolVersion::PROTOCOL_VERSION_2,
+            wep_app->protocol_version());
+  EXPECT_EQ(APIVersion::kAPIV2, wep_app->version().min_supported_api_version);
+  EXPECT_EQ(APIVersion::kAPIV6, wep_app->version().max_supported_api_version);
+
+  const WindowID main_window_id = 0;
+  // Initial HMI level, audio, video streaming states and system context
+  // set after app registration
+  EXPECT_EQ(mobile_apis::HMILevel::INVALID_ENUM,
+            wep_app->hmi_level(main_window_id));
+  EXPECT_EQ(mobile_apis::VideoStreamingState::INVALID_ENUM,
+            wep_app->video_streaming_state());
+  EXPECT_EQ(mobile_apis::AudioStreamingState::INVALID_ENUM,
+            wep_app->audio_streaming_state());
+  EXPECT_EQ(mobile_apis::SystemContext::SYSCTXT_MAIN,
```

Comment:
@AByzhynar I am thinking about moving this checks to a separate function as I see it at least 4 times in your test. However I don't really like a function which accepts app ptr + 4 more input params. 
Probably you may create a separate function `MakeHmiState` which accepts level, a/v state, context and returns `HmiHmiStatePtr` with all configured fields. And another function `CompareApplicationHmiState` which accepts app ptr and hmi state ptr and do the comparison like that. Such approach may make your test fixture shorter and simpler

Reply:
@AKalinich-Luxoft Done in e570bb7a3d5d48b9e64a5172d70804c532ec880f